Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -3142,18 +2045,19 @@ private void setActivityLifeCycleObserver(Application application) {
 
         @Override
         public void onActivityCreated(Activity activity, Bundle bundle) {
+            intentState_ = INTENT_STATE.PENDING;
             if (BranchViewHandler.getInstance().isInstallOrOpenBranchViewPending(activity.getApplicationContext())) {
                 BranchViewHandler.getInstance().showPendingBranchView(activity);
             }
         }
 
         @Override
         public void onActivityStarted(Activity activity) {
+            intentState_ = INTENT_STATE.PENDING;
```

Comment:
For the test bed App the additional delay this adds seems to be `2 - 3 msec`. 
(For TestBed app there is no operation carried out in `onStart()` other than Branch#InitSession)